src/Components/Login.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component }from 'react'
+import LoginForm from './LoginForm'
+import { Button, Message } from 'semantic-ui-react'
+import { signInUser } from '../state/actions/reduxTokenAuthConfig'
+import { connect } from 'react-redux'
+
+class Login extends Component {
+  state = {
+    renderLoginForm: false,
+    email: '',
+    password: '',
+    responseMessage: ''
+  }
+
+  renderFormState = () => {
+    this.setState({
+      renderLoginForm: !this.state.renderLoginForm
+    })
+  }
+
+  inputChangeHandler = (e) => {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+  
+  handleLogin = () => {
+    const { signInUser } = this.props;
+    const { email, password } = this.state;
+      signInUser({ email, password })
+        .then(
+          console.log('Yaaaaaaay')
+        )
+        .catch(error => {
+          this.setState({responseMessage: error.response.data.errors})
+        })
+  }
+
+  render(){
+    let renderLogin
+    let welcomeMessage
+    let responseMessage
+
+    if (this.props.currentUser.isSignedIn) {
+      welcomeMessage = <Message> <h3 id="welcome-message">Hello {this.props.currentUser.attributes.email}</h3></Message>
+    } 
+
+    if(this.state.responseMessage !== ''){
+      responseMessage = <p id="error-message">{this.state.responseMessage}</p>
+    }
+
+    if (this.state.renderLoginForm === true) {
```

Comment:
should these elements be visible if the user is signed in?

Reply:
I'm not quite sure what you mean here? Are you talking about line 44-45 that displays a welcome message once the user is logged in?